crypto/pem/pem_lib.c

```diff
@@ -331,7 +331,7 @@ int PEM_ASN1_write_bio(i2d_of_void *i2d, const char *name, BIO *bp,
 
     if (enc != NULL) {
         objstr = EVP_CIPHER_get0_name(enc);
-        if (objstr == NULL || EVP_CIPHER_get_iv_length(enc) == 0
+        if (objstr == NULL || EVP_CIPHER_get_iv_length(enc) <= 0
```

Comment:
Please don't do this.

Reply:
As I noted earlier, while block_size can have zero as an error, since not cipher returns a block size of zero, the same is not true of iv_len.  An iv length of 0 is a perfectly valid value to return, and as such we have to support returning a value of -1 (or some other invalid value) to indicate error.  The above check is perfectly reasonable.